lib/pbench/test/unit/server/test_requests.py

```diff
@@ -82,7 +82,7 @@ def test_bad_request(client, caplog, server_config, requests_mock, pbench_token)
             headers={"Authorization": "Bearer " + pbench_token},
             json={"indices": "some_invalid_url", "payload": '{ "babble": "42" }'},
         )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert response.status_code == HTTPStatus.BAD_GATEWAY
```

Comment:
Are you sure that this should be `BAD_GATEWAY`?  Yeah, we got a bad status from a RESTful interface that we were calling on behalf of the requestor, but doesn't that represent either a configuration failure on our part or a bad request to us?  (I mean, I can see that there might be times when `BAD_GATEWAY` is appropriate, like when the ES server is choking to death, but it seems like we might want to do some triage, and if the ES server is returning `BAD_REQUEST`, it seems like maybe we should be sending _that_ back to the requestor...eh?)

Reply:
> so far as we know, we did everything right; it's not a bad user request, either.

If that's the case, then I'm OK with `BAD_GATEWAY`.  I just wasn't convinced that that was the case -- in particular, how do we know that it wasn't a bad user request?  I figure that, at the least, given that we think we did everything right, if ES responds `BAD_REQUEST` then we shouldn't be telling the user `BAD_GATEWAY`....

Reply:
Finally looping back to this one, and now I'm on the edge. I've generally thought in terms of our native query APIs where we're transforming the user query into an ES query that's "mostly hidden" from the user, and returning the ES status is not necessarily appropriate.

In this case, however... well, first, I was only fixing the *test* to conform to the existing code, but since this is the `/api/v1/elasticsearch` passthrough query, you're probably correct that we should be returning the Elasticsearch response. Then again, that's expanding the scope of this PR, although in practice the change is probably no bigger than fixing the test and we have no external consumers (and certainly none relying on a failure status).

Reply:
And ... I looked at the code after posting that, and I realize that the `BAD_GATEWAY` is from the common `_call` rather than specific to the passthrough subclass, so it'd require more to fix than just changing a return. That's beyond the scope of this PR, so we should look at the handling of Elasticsearch response status separately.

I still don't know why this test case only started failing for me, or whether it would somehow succeed in Mr Jenkins without this change; but the change matches the deliberate API code, and for now I'm going to leave it at that.